src/components/TrialBanner.js

```diff
@@ -0,0 +1,196 @@
+        ])
+      ])
+    ])
+  }
+
+  render() {
+    const { authState: { isSignedIn, profile }, ajax: { User } } = _.omit('isVisible', this.props)
+    const { accessingCredits, loading, finalizeTrial } = this.state
+    const { trialState } = profile
+    if (!trialState || !isSignedIn || trialState === 'Finalized') return null
+    const { [trialState]: { title, message, enabledLink, button, isWarning } } = messages
+
+    return div([
+      div({
+        style: {
+          display: 'flex', alignItems: 'center', padding: '1.5rem', height: 110,
+          backgroundColor: isWarning ? colors.orange[0] : '#359448',
+          justifyContent: 'center', color: 'white', width: '100%', fontSize: '1rem'
+
+        }
+      },
+      [
+        div({
+          style: {
+            fontSize: '1.5rem', fontWeight: 500, textAlign: 'right', borderRight: '1px solid', paddingRight: '1rem', marginRight: '1rem',
+            maxWidth: 200, flexShrink: 0
+          }
+        }, title),
+        span({ style: { maxWidth: 600, lineHeight: '1.5rem' } },
+          [
+            message,
+            enabledLink && a({
+              style: { textDecoration: 'underline', marginLeft: '0.5rem' },
+              target: '_blank',
+              href: enabledLink.url
+            }, [enabledLink.label, icon('pop-out', { style: { marginLeft: '0.25rem' } })])
+          ]),
+        h(Clickable, {
+          style: {
+            fontWeight: 500, fontSize: '1.125rem', border: '2px solid', borderRadius: '0.25rem', padding: '0.5rem 1rem',
+            marginLeft: '0.5rem', flexShrink: 0
+          },
+          onClick: () => {
+            button.isExternal ? window.open(button.url, '_blank') : this.setState({ accessingCredits: true })
+          }
+        }, [
+          loading ? spinner({ style: { fontSize: '1rem', color: 'white' } }) : button.label,
+          button.isExternal ? icon('pop-out', { style: { marginLeft: '0.25rem' } }) : null
+        ]),
+        div({ style: { marginLeft: '3rem', display: 'flex', flexDirection: 'column', alignItems: 'flex-end' } }, [
+          (trialState === 'Terminated') && h(Clickable, {
+            style: { borderBottom: 'none' },
+            tooltip: 'Hide forever?',
+            onClick: () => this.setState({ finalizeTrial: true })
+          }, [icon('times-circle', { size: 25, style: { fontSize: '1.5rem', cursor: 'pointer', strokeWidth: 1.5 } })])
+        ])
+      ]),
+      accessingCredits && this.renderFreeCreditModal(),
+      finalizeTrial && h(Modal, {
+        title: 'Are you sure?',
+        onDismiss: () => this.setState({ finalizeTrial: false }),
+        okButton: buttonPrimary({
+          onClick: async () => {
+            try {
+              await User.finalizeTrial()
+            } catch (error) {
+              reportError('Error finalizing trial', error)
+            }
+          }
+        }, ['Finalize Trial'])
+      }, ['Finalizing your trial will remove this banner forever.'])
```

Comment:
"Finalize trial" seems unclear, maybe title it "Remove banner" and describe it as "Click confirm to remove this banner". Someone's trial is already over when they see this modal, there isn't anything they can do to un-finalize their trial. The only action they are taking is to remove the banner. 
Is this the same wording as FireCloud or did it come from Comms?

Reply:
In FC they just have the button for hide forever, but I thought the modal would make it a bit more obvious. But I think in practice, the only thing finalizing truly does is remove the banner forever so I'll change the language to reflect that.